core/retries-spi/src/main/java/software/amazon/awssdk/retries/api/RetryStrategy.java

```diff
@@ -198,12 +198,24 @@ default B retryOnRootCauseInstanceOf(Class<? extends Throwable> throwable) {
         B maxAttempts(int maxAttempts);
 
         /**
-         * Configure the backoff strategy used by this executor.
+         * Configure the backoff strategy used by the retry strategy.
          *
          * <p>By default, this uses jittered exponential backoff.
          */
         B backoffStrategy(BackoffStrategy backoffStrategy);
 
+        /**
+         * Configure the backoff strategy used for throttling exceptions by this strategy.
+         *
+         * <p>By default, this uses jittered exponential backoff.
+         */
+        B throttlingBackoffStrategy(BackoffStrategy throttlingBackoffStrategy);
+
+        /**
+         * Configure the predicate to allow the strategy categorize a Throwable as throttling exception.
+         */
+        B treatAsThrottling(Predicate<Throwable> treatAsThrottling);
```

Comment:
Can we clarify in the docs that this will only get called if the `Throwable` is considered retryable? Right now it's not clear when this will be used.

Maybe 
> Configure a predicate that determines whether a retryable exception is a throttling exception. When this predicate returns true, the retry strategy will use the `throttlingBackoffStrategy`. If it returns false, the `backoffStrategy` will be used. This predicate will not be called for non-retryable exceptions.